engine/src/main/java/org/terasology/logic/debug/MovementDebugCommands.java

```diff
@@ -233,22 +231,44 @@ public String stepHeight(@Sender EntityRef client, @CommandParam("height") float
         return "";
     }
 
+    private float function1(float parameter){
+        double p = (double)parameter;
+        return (float) pow(p,0.6) * 0.78f ;
+    }
+
+    private float function2(float parameter){
+       return 0.15f + (0.64f * parameter) - (0.006f * parameter * parameter);
+    }
+
     @Command(shortDescription = "Sets the height of the player", runOnServer = true,
             requiredPermission = PermissionManager.CHEAT_PERMISSION)
     public String playerHeight(@Sender EntityRef client, @CommandParam("height") float amount) {
         try {
             ClientComponent clientComp = client.getComponent(ClientComponent.class);
             CharacterMovementComponent move = clientComp.character.getComponent(CharacterMovementComponent.class);
-            if (move != null) {
+            CharacterComponent charComp=clientComp.character.getComponent(CharacterComponent.class);
+            EntityRef player = client.getComponent(ClientComponent.class).character;
+            GazeMountPointComponent gazeMountPointComponent = player.getComponent(GazeMountPointComponent.class);
+
+            float ratio = amount / 1.6f;
+            float constant= function1(amount);
+
+            if (move != null && gazeMountPointComponent != null) {
                 float prevHeight = move.height;
+
                 move.height = amount;
+                gazeMountPointComponent.translate.y = amount;
+                move.jumpSpeed = 12 * constant;
+                move.stepHeight = 0.35f * ratio;
+                move.distanceBetweenFootsteps = ratio;
+                move.runFactor = function2(amount);
```

Comment:
If `function2` computes the run factor (whatever that is) you could call it `getRunFactor` instead.